src/components/nvml/linux-nvml.c

```diff
@@ -495,6 +527,19 @@ nvml_hardware_read(long long *value, int which_one)
     case FEATURE_POWER:
         *value = getPowerUsage(handle);
         break;
+    case FEATURE_ENERGY_CONSUMPTION:
+        *value = getEnergyConsumption(handle);
+        break;
+    case FEATURE_GPU_INST:
+        field_value->fieldId = NVML_FI_DEV_POWER_INSTANT;
+        field_value->scopeId = 0; // GPU-only scope
+        *value = getDeviceFieldValue(handle, value_count, field_value);
+        break;
+    case FEATURE_GPU_MEMORY_AVG:
+        field_value->fieldId = NVML_FI_DEV_POWER_AVERAGE;
+        field_value->scopeId = 2; // Module-wide scope
```

Comment:
From `nvml.h` I am only seeing a max `scopeId` of 1 which refers to a module scope:
```
 2490  * scopeId needs to be specified. It signifies:
 2491  * 0 - GPU Only Scope - Metrics for GPU are retrieved
 2492  * 1 - Module scope - Metrics for the module (e.g. CPU + GPU) are retrieved.
 2493  * Note: CPU here refers to NVIDIA CPU (e.g. Grace). x86 or non-NVIDIA ARM is not supported
 2494  */
```

I believe that the `scopeId` should be 0 here as well though. As the description mentions "GPU power averaged over 1 sec interval":
```
#define NVML_FI_DEV_POWER_AVERAGE                     185 //!< GPU power averaged over 1 sec interval, supported on Ampere (except GA100) or newer architectures.
```